src/webots/nodes/utils/WbSupervisorUtilities.cpp

```diff
@@ -1003,6 +1104,13 @@ void WbSupervisorUtilities::handleMessage(QDataStream &stream) {
         WbNodeOperations::instance()->importNode(nodeId, fieldId, index, "", nodeString, &importedNodesNumber, true);
       if (operationResult != WbNodeOperations::FAILURE)
         mImportedNodesNumber = importedNodesNumber;
+      const WbField *field = WbNode::findNode(nodeId)->field(fieldId);
+      const WbSFNode *sfNode = dynamic_cast<WbSFNode *>(field->value());
+      if (sfNode) {
+        mImportedNodesNumber = -1;
```

Comment:
it would be better to move this in the else block of the if condition to avoid a double variable set